lib/process-services-cloud/src/lib/task/task-filters/components/edit-task-filter-cloud.component.ts

```diff
@@ -357,11 +357,15 @@ export class EditTaskFilterCloudComponent implements OnInit, OnChanges, OnDestro
     delete(deleteAction: TaskFilterAction): void {
         this.taskFilterCloudService
             .deleteFilter(this.taskFilter)
-            .pipe(takeUntil(this.onDestroy$))
-            .subscribe(() => {
-                deleteAction.filter = this.taskFilter;
-                this.action.emit(deleteAction);
-            });
+            .pipe(
+                filter((filters) => {
+                    deleteAction.filter = this.taskFilter;
+                    this.action.emit(deleteAction);
+                    return filters.length === 0;
```

Comment:
any reason why?

Reply:
Filter will make sure call restore API only when all filters get removed otherwise it will skip calling restore filter API and complete previous call.

Reply:
I don get this `filters.length === 0;`